Driver: Web - "fyne serve" GOOS = Windows

### Checklist

- [X] I have searched the [issue tracker](https://github.com/fyne-io/fyne/issues) for open issues that relate to the same problem, before opening a new one.
- [X] This issue only relates to a single bug. I will open new issues for any other problems.

### Describe the bug

I tried to execute the fyne_demo with "fyne serve" but got following message:
GOOS is not supported, the supported GOOS values are linux and darwin. The GopherJS is falling back to GOOS=linux
panic: open c:\go\src\runtime/fastrand.go: open /\src\runtime/fastrand.go: file does not exist

Does this mean that development of a web version is only possible on Linux and MacOS? 

### How to reproduce

on Windows
>cd \fyne\cmd\fyne_demo
>fyne serve
see the error

### Screenshots

_No response_

### Example code

...

### Fyne version

2.2.2

### Go compiler version

1.17

### Operating system

Windows

### Operating system version

Windows 10

### Additional Information

--

--- a/cmd/fyne/internal/commands/build.go
+++ b/cmd/fyne/internal/commands/build.go
@@ -2,6 +2,7 @@ package commands
 
 import (
 	"encoding/json"
+	"errors"
 	"fmt"
 	"os"
 	"path/filepath"
@@ -171,6 +172,10 @@ func (b *Builder) build() error {
 	goos := b.os
 	if goos == "" {
 		goos = targetOS()
+	}
+
+	if goos == "gopherjs" && runtime.GOOS == "windows" {
+		return errors.New("gopherjs doesn't support Windows. Only wasm target is supported for the web output. You can also use fyne-cross to solve this")
 	}
 
 	fyneGoModRunner := b.runner

--- a/cmd/fyne/internal/commands/build_test.go
+++ b/cmd/fyne/internal/commands/build_test.go
@@ -2,6 +2,7 @@ package commands
 
 import (
 	"fmt"
+	"runtime"
 	"testing"
 
 	"github.com/mcuadros/go-version"
@@ -164,39 +165,47 @@ func Test_BuildWasmReleaseVersion(t *testing.T) {
 }
 
 func Test_BuildGopherJSReleaseVersion(t *testing.T) {
-	expected := []mockRunner{
-		{
-			expectedValue: expectedValue{args: []string{"mod", "edit", "-json"}},
-			mockReturn: mockReturn{
-				ret: []byte("{ \"Module\": { \"Path\": \"fyne.io/fyne/v2\"} }"),
+	expected := []mockRunner{}
+
+	if runtime.GOOS != "windows" {
+		expected = []mockRunner{
+			{
+				expectedValue: expectedValue{args: []string{"mod", "edit", "-json"}},
+				mockReturn: mockReturn{
+					ret: []byte("{ \"Module\": { \"Path\": \"fyne.io/fyne/v2\"} }"),
+				},
 			},
-		},
-		{
-			expectedValue: expectedValue{
-				args:  []string{"version"},
-				osEnv: true,
+			{
+				expectedValue: expectedValue{
+					args:  []string{"version"},
+					osEnv: true,
+				},
+				mockReturn: mockReturn{
+					ret: []byte(""),
+					err: nil,
+				},
 			},
-			mockReturn: mockReturn{
-				ret: []byte(""),
-				err: nil,
+			{
+				expectedValue: expectedValue{
+					args:  []string{"build", "--tags", "release"},
+					osEnv: true,
+					dir:   "myTest",
+				},
+				mockReturn: mockReturn{
+					ret: []byte(""),
+				},
 			},
-		},
-		{
-			expectedValue: expectedValue{
-				args:  []string{"build", "--tags", "release"},
-				osEnv: true,
-				dir:   "myTest",
-			},
-			mockReturn: mockReturn{
-				ret: []byte(""),
-			},
-		},
+		}
 	}
 
 	gopherJSBuildTest := &testCommandRuns{runs: expected, t: t}
 	b := &Builder{appData: &appData{}, os: "gopherjs", srcdir: "myTest", release: true, runner: gopherJSBuildTest}
 	err := b.build()
-	assert.Nil(t, err)
+	if runtime.GOOS == "windows" {
+		assert.NotNil(t, err)
+	} else {
+		assert.Nil(t, err)
+	}
 	gopherJSBuildTest.verifyExpectation()
 }
 

--- a/cmd/fyne/internal/commands/package.go
+++ b/cmd/fyne/internal/commands/package.go
@@ -47,7 +47,7 @@ func Package() *cli.Command {
 			&cli.StringFlag{
 				Name:        "target",
 				Aliases:     []string{"os"},
-				Usage:       "The mobile platform to target (android, android/arm, android/arm64, android/amd64, android/386, ios, iossimulator).",
+				Usage:       "The mobile platform to target (android, android/arm, android/arm64, android/amd64, android/386, ios, iossimulator, wasm, gopherjs, web).",
 				Destination: &p.os,
 			},
 			&cli.StringFlag{

--- a/cmd/fyne/internal/commands/package.go
+++ b/cmd/fyne/internal/commands/package.go
@@ -11,6 +11,7 @@ import (
 	"log"
 	"os"
 	"path/filepath"
+	"runtime"
 	"strconv"
 	"strings"
 
@@ -227,6 +228,10 @@ func (p *Packager) buildPackage(runner runner) ([]string, error) {
 	err := bWasm.build()
 	if err != nil {
 		return nil, err
+	}
+
+	if runtime.GOOS == "windows" {
+		return []string{bWasm.target}, nil
 	}
 
 	bGopherJS := &Builder{

--- a/cmd/fyne/internal/commands/package_test.go
+++ b/cmd/fyne/internal/commands/package_test.go
@@ -477,7 +477,11 @@ func Test_BuildPackageWeb(t *testing.T) {
 	files, err := p.buildPackage(webBuildTest)
 	assert.Nil(t, err)
 	assert.NotNil(t, files)
-	assert.Equal(t, 2, len(files))
+	expectedFiles := 2
+	if runtime.GOOS == "windows" {
+		expectedFiles = 1
+	}
+	assert.Equal(t, expectedFiles, len(files))
 }
 
 func Test_PackageWeb(t *testing.T) {

--- a/cmd/fyne/internal/commands/package_test.go
+++ b/cmd/fyne/internal/commands/package_test.go
@@ -234,6 +234,9 @@ func Test_PackageWasm(t *testing.T) {
 }
 
 func Test_buildPackageGopherJS(t *testing.T) {
+	if runtime.GOOS == "windows" {
+		t.Skip()
+	}
 	expected := []mockRunner{
 		{
 			expectedValue: expectedValue{args: []string{"mod", "edit", "-json"}},
@@ -292,6 +295,9 @@ func Test_buildPackageGopherJS(t *testing.T) {
 }
 
 func Test_PackageGopherJS(t *testing.T) {
+	if runtime.GOOS == "windows" {
+		t.Skip()
+	}
 	expected := []mockRunner{
 		{
 			expectedValue: expectedValue{args: []string{"mod", "edit", "-json"}},
@@ -499,33 +505,38 @@ func Test_PackageWeb(t *testing.T) {
 				ret: []byte(""),
 			},
 		},
-		{
-			expectedValue: expectedValue{args: []string{"mod", "edit", "-json"}},
-			mockReturn: mockReturn{
-				ret: []byte("{ \"Module\": { \"Path\": \"fyne.io/fyne/v2\"} }"),
+	}
+
+	if runtime.GOOS != "windows" {
+		expected = append(expected, []mockRunner{
+			{
+				expectedValue: expectedValue{args: []string{"mod", "edit", "-json"}},
+				mockReturn: mockReturn{
+					ret: []byte("{ \"Module\": { \"Path\": \"fyne.io/fyne/v2\"} }"),
+				},
 			},
-		},
-		{
-			expectedValue: expectedValue{
-				args:  []string{"version"},
-				osEnv: true,
+			{
+				expectedValue: expectedValue{
+					args:  []string{"version"},
+					osEnv: true,
+				},
+				mockReturn: mockReturn{
+					ret: []byte(""),
+					err: nil,
+				},
 			},
-			mockReturn: mockReturn{
-				ret: []byte(""),
-				err: nil,
+			{
+				expectedValue: expectedValue{
+					args: []string{"build",
+						"-o", "myTest.js"},
+					osEnv: true,
+					dir:   "myTest",
+				},
+				mockReturn: mockReturn{
+					ret: []byte(""),
+				},
 			},
-		},
-		{
-			expectedValue: expectedValue{
-				args: []string{"build",
-					"-o", "myTest.js"},
-				osEnv: true,
-				dir:   "myTest",
-			},
-			mockReturn: mockReturn{
-				ret: []byte(""),
-			},
-		},
+		}...)
 	}
 
 	p := &Packager{

--- a/cmd/fyne/internal/commands/serve.go
+++ b/cmd/fyne/internal/commands/serve.go
@@ -4,6 +4,7 @@ import (
 	"errors"
 	"fmt"
 	"net/http"
+	"runtime"
 	"strconv"
 
 	"github.com/urfave/cli/v2"
@@ -19,6 +20,11 @@ type Server struct {
 // Serve return the cli command for serving fyne wasm application over http
 func Serve() *cli.Command {
 	s := &Server{appData: &appData{}}
+
+	target := "web"
+	if runtime.GOOS == "windows" {
+		target = "wasm"
+	}
 
 	return &cli.Command{
 		Name:        "serve",
@@ -46,7 +52,7 @@ func Serve() *cli.Command {
 				Name:        "target",
 				Aliases:     []string{"os"},
 				Usage:       "The web runtime to target (wasm, gopherjs, web).",
-				Value:       "web",
+				Value:       target,
 				Destination: &s.os,
 			},
 		},

